src/lib/dapp/index.ts

```diff
@@ -180,16 +163,18 @@ export class DAppConnector {
    * @param pairingTopic - The pairing topic for the connection (optional).
    * @returns A Promise that resolves when the connection process is complete.
    */
-  public async openModal(pairingTopic?: string): Promise<void> {
-    const { uri, approval } = await this.connectURI(pairingTopic)
-    const walletConnectModal = new WalletConnectModal({
-      projectId: this.projectId,
-      chains: this.supportedChains,
-    })
-    walletConnectModal.openModal({ uri })
-    const session = await approval()
-    await this.onSessionConnected(session)
-    walletConnectModal.closeModal()
+  public async openModal(pairingTopic?: string): Promise<SessionTypes.Struct> {
```

Comment:
Can you update the jsdoc's return type to reflect that SessionType.Struct is returned now?